src/common/useCustomCursor.ts

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+
+export default function useCustomCursor(
+    ref: React.RefObject<HTMLElement> | null,
+    cursorTemplate: string,
+    color: string,
+    offsetX = 0,
+    offsetY = 0,
+    defaultCursor = 'default',
+): void {
+    React.useEffect(() => {
+        const { current: element } = ref || {};
+
+        if (!element) {
+            return;
+        }
+
+        let svg = cursorTemplate.replace('{{cursorFillColor}}', color);
+        // convert svg to data-uri
+        svg = encodeURIComponent(svg.replace(/"/g, "'"));
```

Comment:
Can we avoid reassignment by chaining the `.replace` calls and wrapping them in `encodeURIComponent`?

```
const svg = encodeURIComponent(
    cursorTemplate
        .replace('{{cursorFillColor}}', color)
        .replace(/"/g, "'")
);